### PR TITLE
Missing testsuite

### PR DESCRIPTION
**Pull request by Algatux**
**Proposal for adding a test suite**

added: 
- travis configuration testing against php >= 5.4 including symfony to verify conficts
- phpunit configuration and dev-dependency
- a simple and incomplete test to start

we must create a travis accunt and enable the repo if this well be merged

### CODE REVIEW

**Comment by Jean85** [comment]
Great!! Nice to see this ;)

I have just one question: why did you choose SF 2.8 and 3.0 as version to be tested?
2.8 is the latest LTS, and that's ok, but should we test 3.1 now, since it's the latest stable release?

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  - 5.4
+  - 5.5
+  - 5.6
+  - 7.0
+
+# Commands to be run before your environment runs.
+before_script:
+  - composer self-update
+  - if [ "$SYMFONY" != "" ]; then composer require "symfony/symfony:${SYMFONY}" --no-update; fi;
+  - composer update --prefer-source --no-interaction ${COMPOSER_FLAGS}
```
Comment (Jean85) [inline .travis.yml]: Why `--prefer-source`? Isn't `--prefer-dist` faster?

File: composer.json [hunk 2]
```diff
@@ -6,6 +6,9 @@
         "php" : ">=5.4",
         "doctrine/dbal": ">=2.3,<2.6-dev"
     },
+    "require-dev": {
+        "phpunit/phpunit": "4.8"
```
Comment (Jean85) [inline composer.json]: Shouldn't `@stable` be better? Or at least a easier constraint?

**Comment by Algatux** [comment]
@Jean85 
> I have just one question: why did you choose SF 2.8 and 3.0 as version to be tested?
> 2.8 is the latest LTS, and that's ok, but should we test 3.1 now, since it's the latest stable release?

3.1 now has an issue with prophecy latest release we will add it when it is solved.

> Why --prefer-source? Isn't --prefer-dist faster?
will investigate but yes, dist it's faster

> Shouldn't @stable be better? Or at least a easier constraint?
I choosed 4.8 to mantain compatibility with php 5.4

**Comment by Jean85** [comment]
The 5.4 compatibility for PHPUnit should be a concern resolved by Composer: with `@stable` it will install 4.8 on PHP >= 5.5, and 5.x on PHP >= 5.6

**Comment by thomasvargiu** [comment]
Why require an huge framework for tests?

I don't think this is necessary. It's something related to Doctrine, not to Symfony.

**Comment by Algatux** [comment]
> Why require an huge framework for tests?
> I don't think this is necessary. It's something related to Doctrine, not to Symfony.

requiring symfony or zf will check for packages constraint compatibility. with this trick for example I found an issue with prophecy and symfony 3.1.

yes the package is not strictly related to a framework but like the readme suggests it's thougth to work with (IMHO)

File: composer.json (hunk 2 above)
Comment (thomasvargiu) [inline composer.json]: For testing `--prefer-lowest` I think it's better to use:
`"phpunit/phpunit": "^4.8 || ^5.0"`

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,61 @@
+# Required to run your project under the correct environment.
+language: php
+
+# Versions of PHP you want your project run with.
+php:
```
Comment (thomasvargiu) [inline .travis.yml]: The `php` key is not necessary, they are included in the `matrix` key.

File: composer.json [hunk 4]
```diff
@@ -14,7 +17,14 @@
         }
     ],
     "autoload": {
-        "psr-4": { "Facile\\DoctrineMySQLComeBack\\Doctrine\\DBAL\\": "src/" }
+        "psr-4": { "Facile\\DoctrineMySQLComeBack\\Doctrine\\DBAL\\": "src/" },
+        "exclude-from-classmap": [
```
Comment (thomasvargiu) [inline composer.json]: Tests should not be included in the classmap, but should be better to add it in `autoload-dev`

**Comment by thomasvargiu** [comment]
@Algatux 
Tests are ok. I suggest to:
- Make coverage just for one PHP version.
- Disable `xdebug` for the others PHP versions.
- Push the coverage to scrutinizer.

An example in another project could be found [here](https://github.com/facile-it/sentry-module/blob/master/.travis.yml).

**Comment by Algatux** [comment]
Fixed .travis.yml and composer.json

**Comment by peelandsee** [comment]
love it @Algatux , thanks!

**Pull request merged**